script.js

```diff
@@ -1,4 +1,162 @@
+// Global Variables
+var GAME_STATE_DICE_ROLL = "GAME_STATE_DICE_ROLL";
+var GAME_STATE_CHOOSE_DICE_ORDER = "GAME_STATE_CHOOSE_DICE_ORDER";
+var GAME_STATE_COMPARE_SCORES = "GAME_STATE_COMPARE_SCORES";
+var gameState = GAME_STATE_DICE_ROLL;
+
+var currentPlayerRolls = [];
+
+var currentPlayer = 1;
+var allPlayersScore = [];
+
+// Helper Function
+var rollDice = function () {
+  console.log("Control flow: start of rollDice()");
```

Comment:
console logs usually are internal checks for your use only, it usually is omitted for final code submissions.